pkg/riff/commands/function_status_test.go

```diff
@@ -0,0 +1,117 @@
+/*
+ * Copyright 2019 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package commands_test
+
+import (
+	"testing"
+
+	duckv1alpha1 "github.com/knative/pkg/apis/duck/v1alpha1"
+	buildv1alpha1 "github.com/projectriff/system/pkg/apis/build/v1alpha1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+
+	"github.com/projectriff/riff/pkg/riff/commands"
+	rifftesting "github.com/projectriff/riff/pkg/testing"
+)
+
+func TestFunctionStatusOptions(t *testing.T) {
+	table := rifftesting.OptionsTable{
+		{
+			Name: "invalid resource",
+			Options: &commands.FunctionStatusOptions{
+				ResourceOptions: rifftesting.InvalidResourceOptions,
+			},
+			ExpectFieldError: rifftesting.InvalidResourceOptionsFieldError,
+		},
+		{
+			Name: "valid resource",
+			Options: &commands.FunctionStatusOptions{
+				ResourceOptions: rifftesting.ValidResourceOptions,
+			},
+			ShouldValidate: true,
+		},
+	}
+
+	table.Run(t)
+}
+
+func TestFunctionStatusCommand(t *testing.T) {
+	defaultNamespace := "default"
+	functionName := "test-function"
+	function := &buildv1alpha1.Function{
+		ObjectMeta: metav1.ObjectMeta{
+			Namespace: defaultNamespace,
+			Name:      functionName,
+		},
+		Spec: buildv1alpha1.FunctionSpec{
+			Image:    "projectriff/upper",
+			Artifact: "uppercase.js",
+			Handler:  "functions.Uppercase",
+		},
+		Status: buildv1alpha1.FunctionStatus{
+			Status: duckv1alpha1.Status{
+				Conditions: []duckv1alpha1.Condition{
+					{
+						Type:     buildv1alpha1.FunctionConditionReady,
+						Status:   "True",
+						Reason:   "",
+						Severity: "",
+						Message:  "",
+					},
+				},
+			},
+			BuildStatus: buildv1alpha1.BuildStatus{
+				LatestImage: "projectriff/upper@sah256:abcdef1234",
+			},
+		},
+	}
+
+	table := rifftesting.CommandTable{
+		{
+			Name:        "function does not exist",
+			Args:        []string{functionName},
+			ShouldError: true,
+		},
+		{
+			Name: "show status",
+			Args: []string{functionName},
+			GivenObjects: []runtime.Object{
+				function,
+			},
+			ExpectOutput: `
+FUNCTION:
+ Name: test-function
```

Comment:
a flat collection that mushes together values from ObjectMeta, FunctionSpec and FunctionStatus is a bit messy. `kubectl describe` attempts to make a resource easier to read without hiding the inherent structure of that resource.

Other questions that `riff function status` should be able to easily answer:
- where is the source?
- was a build cache requested?
- is the build cache provisioned?

Reply:
yeah, I'm sure we can add a lot of information here

Reply:
this is just a first try at adding some of this info